AFQ/tasks/tractography.py

```diff
@@ -134,14 +139,62 @@ def streamlines(data_imap, seed, stop,
 
     is_trx = this_tracking_params.get("trx", False)
 
+    num_chunks = this_tracking_params.pop("num_chunks", False)
+
     if is_trx:
         start_time = time()
         dtype_dict = {'positions': np.float16, 'offsets': np.uint32}
-        lazyt = aft.track(params_file, **this_tracking_params)
-        sft = TrxFile.from_lazy_tractogram(
-            lazyt,
-            seed,
-            dtype_dict=dtype_dict)
+        if num_chunks:
+            if not has_ray:
+                raise ImportError("Ray is required to perform tractography in"
+                                  "parallel, please install ray or remove the"
+                                  " 'num_chunks' arg")
+
+            @ray.remote
+            class TractActor():
+                def __init__(self):
+                    self.TrxFile = TrxFile
+                    self.aft = aft
+                    self.objects = {}
+
+                def trx_from_lazy_tractogram(self, lazyt_id, seed, dtype_dict):
+                    id = self.objects[lazyt_id]
+                    return self.TrxFile.from_lazy_tractogram(id,
+                                                             seed,
+                                                             dtype_dict)
+
+                def create_lazyt(self, id, *args, **kwargs):
+                    self.objects[id] = self.aft.track(*args, **kwargs)
+                    return id
+
+                def delete_lazyt(self, id):
+                    if id in self.objects:
+                        del self.objects[id]
+            actors = [TractActor.remote() for _ in range(num_chunks)]
+            object_id = 1
+
+            # create lazyt inside each actor
+            tasks = [ray_actor.create_lazyt.remote(object_id, params_file,
+                     **this_tracking_params) for ray_actor in actors]
+            ray.get(tasks)
+
+            # create trx from lazyt
+            tasks = [ray_actor.trx_from_lazy_tractogram.remote(object_id, seed,
+                     dtype_dict) for ray_actor in actors]
+            sfts = ray.get(tasks)
+
+            # cleanup objects
+            tasks = [ray_actor.delete_lazyt.remote(object_id) for ray_actor in
+                     actors]
+            ray.get(tasks)
+
+            sft = concatenate(sfts)
```

Comment:
For clarity, I would 

```
from trx.trx_file_memmap import concatenate as trx_concat
``` 

at the top and then here call: 

```suggestion
            sft = trx_concat(sfts)
```

Reply:
fixed in lastest push